File: AdBoard.Api/src/main/kotlin/ru/atom/adboard/dal/entities/Category.kt
package ru.atom.adboard.dal.entities
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "categories")
data class Category(
    @Id
    @Column(name = "id", columnDefinition = "UUID")
    val id: UUID = UUID.randomUUID(),

    @Column(name = "name", nullable = false)
    val name: String,

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "category")
    val subcategories:  List<Subcategory> = Collections.emptyList()
)

File: AdBoard.API/AdBoard.DAL/Entities/Category.cs
using System;
using System.Collections.Generic;

namespace AdBoard.DAL.Entities;

public class Category
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public ICollection<Subcategory> Subcategories { get; set; } = new List<Subcategory>();
}
